src/StreamJsonRpc.Shared/Reflection/MethodNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have an alternate name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class MethodNameAttribute : Attribute
+    {
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="name">Alternate name of a method.</param>
```

Comment:
It's meant to replace, so alternate name was a bit misleading.  I'll change that.  If there are overloads with different MethodNameAttribute values, then it would result in an error.  I'll be more clear in the description.  Good case on the overload scenario, I've added a test.